taskcluster/fenix_taskgraph/release_promotion.py

```diff
@@ -133,9 +141,23 @@ def release_promotion_action(parameters, graph_config, input, task_group_id, tas
     parameters['optimize_target_tasks'] = True
     parameters['shipping_phase'] = input['release_promotion_flavor']
 
+    version_in_file = read_version_file()
     parameters['version'] = input['version'] if input.get('version') else read_version_file()
-    parameters['head_tag'] = 'v{}'.format(parameters['version'])
-    parameters['release_type'] = resolve_release_type(parameters['head_tag'])
+    version_string = parameters['version']
+    if version_string != version_in_file:
+        raise ValueError("Version given in tag ({}) does not match the one in version.txt ({})".format(version_string, version_in_file))
+    parameters['head_tag'] = 'v{}'.format(version_string)
+
+    parameters['next_version'] = input['next_version']
+
+    version = FenixVersion.parse(version_string)
+    if version.is_beta:
```

Comment:
I could have factorized this behavior with what we have in `parameters.py`. Although, I intend that logic to be gone by the end of the year.

This is why, I'm okay to have a little bit of code duplication here.

Reply:
Makes sense 👍